crates/bevy_ecs/src/identifier/error.rs

```diff
@@ -0,0 +1,22 @@
+//! Error types for [`super::Identifier`] conversions. An ID can be converted
+//! to various kinds, but these can fail if they are not valid forms of those
+//! kinds. The error type in this module encapsulates the various failure modes.
+use std::fmt;
+
+/// An  Error type for [`super::Identifier`], mostly for providing error
+/// handling for convertions of an ID to a type abstracting over the ID bits.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum IdentifierError {
```

Comment:
Do we want to put `#[non_exhaustive]` on this?

Reply:
I think so. New error variants shouldn't be breaking, as they will only affect code paths that are added in the future, and won't affect current ones.